src/natcap/invest/habitat_quality.py

```diff
@@ -269,200 +295,276 @@ def execute(args):
-                        xoff=block_offset['xoff'])
-                threat_band = None
-                threat_raster = None
+                    aligned_bound_threat_path)
+
+                # Bound threat raster values to 0 <= x <= 1
+                bound_threat_values_task = task_graph.add_task(
+                    _bound_raster_values,
+                    args=(
+                        (aligned_threat_path, 1), aligned_bound_threat_path),
+                    target_path_list=[aligned_bound_threat_path],
+                    dependent_task_list=[align_task],
+                    task_name=f'bound_threat{lulc_key}_{threat}')
+                bound_threat_values_tasks.append(bound_threat_values_task)
 
     LOGGER.info('Starting habitat_quality biophysical calculations')
-
     # Rasterize access vector, if value is null set to 1 (fully accessible),
     # else set to the value according to the ACCESS attribute
     cur_lulc_path = lulc_path_dict['_c']
     fill_value = 1.0
-    try:
-        LOGGER.info('Handling Access Shape')
-        access_raster_path = os.path.join(
-            inter_dir, 'access_layer%s.tif' % suffix)
-        # create a new raster based on the raster info of current land cover
-        pygeoprocessing.new_raster_from_base(
-            cur_lulc_path, access_raster_path, gdal.GDT_Float32,
-            [_OUT_NODATA], fill_value_list=[fill_value])
-        pygeoprocessing.rasterize(
-            args['access_vector_path'], access_raster_path, burn_values=None,
-            option_list=['ATTRIBUTE=ACCESS'])
-
-    except KeyError:
-        LOGGER.info(
-            'No Access Shape Provided, access raster filled with 1s.')
+
+    LOGGER.info('Handling Access Shape')
+    access_raster_path = os.path.join(
+        intermediate_output_dir, f'access_layer{file_suffix}.tif')
+    # create a new raster based on the raster info of current land cover
+    create_access_raster_task = task_graph.add_task(
+        pygeoprocessing.new_raster_from_base,
+        args=(cur_lulc_path, access_raster_path, gdal.GDT_Float32,
+              [_OUT_NODATA]),
+        kwargs={
+            'fill_value_list': [fill_value]
+            },
+        target_path_list=[access_raster_path],
+        dependent_task_list=[align_task],
+        task_name=f'access_raster')
+    access_task_list = [create_access_raster_task]
+
+    if 'access_vector_path' in args:
+        LOGGER.debug("Rasterize Access vector")
+        rasterize_access_task = task_graph.add_task(
+            pygeoprocessing.rasterize,
+            args=(args['access_vector_path'], access_raster_path),
+            kwargs={
+                'option_list': ['ATTRIBUTE=ACCESS'],
+                'burn_values': None
+                },
+            target_path_list=[access_raster_path],
+            dependent_task_list=[create_access_raster_task],
+            task_name=f'rasterize_access')
+        access_task_list.append(rasterize_access_task)
```

Comment:
@phargogh we would need to create that access raster which is defaulted to 1's to pass along to the `_calculate_total_degradation` op. The way `total_degradation` operation is set up in that function, it expects a certain pattern of inputs since we don't know how many threat rasters we might have, so it assumes the last value is the access raster...